### PR TITLE
[Team 9] Add key press to test the spawn of bosses

### PR DESCRIPTION
**Pull request by phurinjeffy**
# Description

Added a key press used for spawning each boss (temporary used for testing, as currently spawning the bosses require you to complete quests):
- Press 1: Spawn land boss
- Press 2: Spawn water boss
- Press 3: Spawn air boss

## Type of change

- New feature (non-breaking change which adds functionality)
- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Passed the unit tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java [hunk 1]
```diff
@@ -122,10 +122,12 @@ public void create() {
       player.getEvents().addListener("defeatLandBoss", this::spawnKangarooBoss);
```
Comment (lellis0406) [inline source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:122]: these events are reserved for quest completion upon boss death, please use a new event name

File: source/core/src/main/com/csse3200/game/components/player/KeyboardPlayerInputComponent.java [hunk 2]
```diff
@@ -92,6 +92,15 @@ public boolean keyDown(int keycode) {
       case Keys.I:
         entity.getEvents().trigger("statsInfo");
         return true;
+      case Keys.NUM_1: // TEMPORARY: Press 1 to spawn the land boss
+        entity.getEvents().trigger("defeatLandBoss");
```
Comment (lellis0406) [inline source/core/src/main/com/csse3200/game/components/player/KeyboardPlayerInputComponent.java:96]: these events are reserved for quest completion upon boss death, see other comment

**Review by jakingy — APPROVED** [review]
Useful for testing functionality relating to the boss, just make sure to fix merge conflict.

**Review by lellis0406 — CHANGES_REQUESTED** [review]
please see changes below to decouple from quest functionality

**Comment by lellis0406** [comment]
looks good

**Review by lellis0406 — APPROVED** [review]
looks good!

**Pull request merged**